consensus/types/src/subnet_id.rs

```diff
@@ -77,48 +76,16 @@ impl SubnetId {
     /// Computes the set of subnets the node should be subscribed to during the current epoch,
     /// along with the first epoch in which these subscriptions are no longer valid.
     #[allow(clippy::arithmetic_side_effects)]
-    pub fn compute_subnets_for_epoch<E: EthSpec>(
+    pub fn compute_attestation_subnets(
         raw_node_id: [u8; 32],
-        epoch: Epoch,
         spec: &ChainSpec,
-    ) -> Result<(impl Iterator<Item = SubnetId>, Epoch), &'static str> {
-        // simplify variable naming
-        let subscription_duration = spec.epochs_per_subnet_subscription;
+    ) -> impl Iterator<Item = SubnetId> {
```

Comment:
Just to confirm my understanding - this change doesn't impact discovery because `compute_attestation_subnets` is only used by the node to compute its own permanent subscriptions, and the node then advertises its persistent attestation subnet via ENR attestation bitfield.

Peers then use attestation bitfield advertised by the node for discovery - which means different nodes could have different logic for computing attestation subnets?

Reply:
Exactly.